plugins/sync.js

```diff
@@ -259,62 +281,71 @@ window.plugin.sync.RegisteredPluginsFields.prototype.add = function(registeredMa
   this.waitingInitialize[registeredMap.getFileName()] = registeredMap;
 
-  this.initializeWorker();
+  this.initializeWorker();                                  // start the regular check on drive
 };
 
 window.plugin.sync.RegisteredPluginsFields.prototype.get = function(pluginName, fieldName) {
   if(!this.pluginsfields[pluginName]) return;
   return this.pluginsfields[pluginName][fieldName];
 };
 
+// try to initialize remaining entries in the queue
 window.plugin.sync.RegisteredPluginsFields.prototype.initializeRegistered = function() {
   var _this = this;
   if(this.authorizer.isAuthed()) {
     $.each(this.waitingInitialize, function(key, map) {
-      if(!map.initializing && !map.initialized) {
-        map.initialize(_this.cleanWaitingInitialize);
+      if(!map.initializing && !map.initialized) {           // if not alraedy in process
+        map.initialize(_this.cleanWaitingInitialize);       // initialize the map
       }
     });
   }
 };
 
+// cleanup the initialization queue
 window.plugin.sync.RegisteredPluginsFields.prototype.cleanWaitingInitialize = function() {
   var newWaitingInitialize, _this;
   _this = this;
 
   newWaitingInitialize = {};
-  $.each(this.waitingInitialize, function(key,map) {
-    if(map.failed) _this.anyFail = true;
-    if(map.initialized || map.failed) return true;
-    newWaitingInitialize[map.getFileName()] = map;
+  $.each(this.waitingInitialize, function(key,map) {        // test queue if all data is ready
+    if(map.failed) _this.anyFail = true;                    // at least one failed ?
+    if(map.initialized || map.failed) return true;          // next map
+    newWaitingInitialize[map.getFileName()] = map;          // copy remaining maps
   });
-  this.waitingInitialize = newWaitingInitialize;
+  this.waitingInitialize = newWaitingInitialize;            // to the queue
 };
 
+// Try to initialize all registerd fields with a timeout of 10 sec
 window.plugin.sync.RegisteredPluginsFields.prototype.initializeWorker = function() {
   var _this = this;
 
-  this.cleanWaitingInitialize();
-  plugin.sync.toggleDialogLink();
-  this.initializeRegistered();
-
-  clearTimeout(this.timer);
-  if(Object.keys(this.waitingInitialize).length > 0) {
-    this.timer = setTimeout(function() {_this.initializeWorker()}, 10000);
+  this.cleanWaitingInitialize();                            // clean up the initialization queue
+  plugin.sync.toggleDialogLink();                           // set the color of the menu entry
+  this.initializeRegistered();                              // try initialize remaining entries 
+                                                            //   in the queue
+  clearTimeout(this.timer);                                 // reset the timer
+  if(Object.keys(this.waitingInitialize).length > 0) {      // if queue still not empty
+    this.timer = setTimeout(
+      function() {_this.initializeWorker()}, 10000          // retry in 10 sec
```

Comment:
move time constant to extra variable
```suggestion
      function() {_this.initializeWorker()}, RETRY_TIME
```

NB: also comment in 318 references this